docs-ref-services/key-vault.md

```diff
@@ -1,138 +1,107 @@
-    tenant = '...'
-)
+To use Managed Identities, acquire a `DefaultAzureCredential` object as shown in the following code. (To run this code on a developer workstation, see [Using Managed Identities locally](#using-managed-identities-locally)).
 
-client = KeyVaultClient(credentials)
+```python
+# Authentication with Managed Identities
+from azure.identity import DefaultAzureCredential
 
-# VAULT_URL must be in the format 'https://<vaultname>.vault.azure.net'
-# KEY_VERSION is required, and can be obtained with the KeyVaultClient.get_key_versions(self, vault_url, key_name) API
-key_bundle = client.get_key(VAULT_URL, KEY_NAME, KEY_VERSION)
-key = key_bundle.key
+# Acquire default Azure credentials, which are useful in most circumstances.
+credentials = DefaultAzureCredential()
 ```
 
-Retrieve a secret from a vault:
+To use a service principal, acquire a `ServicePrincipalCredential` object as shown in the following code. Review [Authenticate with the Azure SDK for Python](/azure/python/python-sdk-azure-authenticate) for information on creating a service principal and obtaining a tenant ID, a client ID, and application (or client) secret using the `az ad sp create-for-rbac --sdk-auth` command.
+
+To find the object ID for the service principal, go to the [Azure portal](https://portal.azure.com), navigate to your subscription, select **Access Control (IAM)**, select **Role assignments**, and locate the appropriate service principal. Select that service principal to open it, then find the object ID on the **Overview** page.
 
 ```python
-from azure.keyvault import KeyVaultClient
+# Authentication with a service principal
 from azure.common.credentials import ServicePrincipalCredentials
 
-credentials = ServicePrincipalCredentials(
-    client_id = '...',
-    secret = '...',
-    tenant = '...'
-)
-
-client = KeyVaultClient(credentials)
-
-# VAULT_URL must be in the format 'https://<vaultname>.vault.azure.net'
-# SECRET_VERSION is required, and can be obtained with the KeyVaultClient.get_secret_versions(self, vault_url, secret_id) API
-secret_bundle = client.get_secret(VAULT_URL, SECRET_ID, SECRET_VERSION)
-secret = secret_bundle.value
-```
-
-> [!div class="nextstepaction"]
-> [Explore the Client APIs](/python/api/overview/azure/keyvault/client)
-
-### Management library
+# Replace the values with your specific IDs
+client_id = "your_client_id"
+tenant_id = "your_tenant_id"
+app_secret = "your_app_secret"
 
-```bash
-pip install azure-mgmt-keyvault
+credentials = ServicePrincipalCredentials(client_id=client_id, secret=app_secret, tenant=tenant_id)
 ```
 
-### Example
-
-The following example shows how to create an Azure Key Vault. 
+With the `credentials` variable initialized, you can then use the following code to create an Azure Key Vault named `msdocs-vault01` in an *existing* resource group named `KeyVault-rg`. With the Key Vault in place, you can then use the [client library](#client-library) to create, retrieve, and delete items.
```

Comment:
Please verify this works using a service principal created as instructed in the other document. The principal needs Azure Resource Manager access to create a Vault.